### ISSUE TITLE
Video pulsating with VP8 and Adaptive

### ISSUE THREAD

**Issue by SelimEmre** [ant-media/Ant-Media-Server#2507]
### Short description
Video pulsating happening with VP8 and Adaptive option enabled.

### Steps to reproduce
1. Add 1 adaptive option and enable VP8  
2. Disable H.264 option.
3. Publish stream into OBS with AMS default encoder settings and [this file](https://drive.google.com/file/d/1iyfIs5gDDI_hD3gbd_uX0dYDvROE9tnT/view?usp=sharing)

### Expected behavior
You should see the normal video.

### Actual behavior
We see pulsating in the video.
 
### Logs
In latest SNAPSHOT version.

**Comment by mekya** [comment]
TL;DR;
Increase the adaptive bitrate further than default ones until you don't see the pulsating


Thank you for the report. I've reproduced this issue on our end. The problem seems to be the encoder related. Maybe it should be called encoder tuning. Let me give some more information.  

The problem directly can be re-produced by ffmpeg command like this. 
```
ffmpeg -i out.mp4 -vcodec libvpx -deadline realtime -y  -g 10  outfreeze.webm
```
The encoder creates this pulsating effect whenever it creates the I-frame so that if you increase the gop size you see less pulsating. Of course this is not a solution. Secondly, we've used the `-deadline realtime` parameter above because Ant Media Server uses `realtime` by default.  

On the other hand, if you use the below command, there is almost no pulsating.
```
ffmpeg -i out.mp4 -vcodec libvpx -deadline realtime -b:v 4M -crf 23 -g 25  -y out.webm
```

If you want to not increase bitrate and want to tune parameters, here are the things to try. 

You can change the deadline parameter to `good` or `best` through app configuraiton as shown below
 ```
settings.encoding.vp8.deadline=good
```

Changing deadline to good or best absolutely increases the quality however there is a trade-off in the speed. This causes audio & video synch issue. So you need to tune the speed of encoder with following property. 
```
 settings.encoding.vp8.speed=4
```
This property gets values between -16 and 16. 
-16 means slowest and best quality, 16 means fastest and worst quality for that specific deadline. 

Maybe the last parameter is the thread count for the encoder.  You can configure it via following property
```
settings.encoding.vp8.threadCount=4
```

Also don't forget to change the bitrate parameter through web panel while testing. By tuning these parameters, you can catch your acceptable quality by still having real time streaming.
